src/lib/eco-router/api.ts

```diff
@@ -258,164 +191,91 @@ export async function getExactOut(
+            quoteCurrency: currencyIn,
+            amount: currencyAmountOut,
+            maximumSlippage,
+            recipient: receiver,
+            tradeType: TradeType.EXACT_OUTPUT,
+          })
+        }
+        // Curve
+        if (platform.name === RoutablePlatform.CURVE.name) {
+          return CurveTrade.bestTradeExactOut({
+            currencyAmountOut,
+            currencyIn,
+            maximumSlippage,
+            receiver,
+          })
+        }
+        // ZeroX
+        if (platform.name === RoutablePlatform.ZEROX.name) {
+          return ZeroXTrade.bestTradeExactOut(currencyIn, currencyAmountOut, maximumSlippage)
+        }
+        // COW
+        if (platform.name === RoutablePlatform.COW.name) {
+          return CoWTrade.bestTradeExactOut({
+            currencyAmountOut,
+            currencyIn,
+            maximumSlippage,
+            receiver,
+            user,
+          })
+        }
+        if (platform.name === RoutablePlatform.VELODROME.name) {
+          return VelodromeTrade.getQuote({
+            quoteCurrency: currencyIn,
+            amount: currencyAmountOut,
+            maximumSlippage,
+            recipient: receiver,
+            tradeType: TradeType.EXACT_OUTPUT,
+          })
+        }
+        if (platform.name === RoutablePlatform.ONE_INCH.name) {
+          return OneInchTrade.getQuote({
+            quoteCurrency: currencyIn,
+            amount: currencyAmountOut,
+            maximumSlippage,
+            recipient: receiver,
+            tradeType: TradeType.EXACT_OUTPUT,
+          })
+        }
+      } catch (error) {
         errors.push(error)
-        resolve(undefined)
-      })
-  })
-
-  // Curve
-  const curveTrade = new Promise<CurveTrade | undefined>(async resolve => {
-    if (!RoutablePlatform.CURVE.supportsChain(chainId)) {
-      return resolve(undefined)
-    }
-
-    CurveTrade.bestTradeExactOut({
-      currencyAmountOut,
-      currencyIn,
-      maximumSlippage,
-      receiver,
-    })
-      .then(resolve)
-      .catch(error => {
-        errors.push(error)
-        resolve(undefined)
-      })
-  })
-
-  // ZeroX
-  const zeroXTrade = new Promise<ZeroXTrade | undefined>(async resolve => {
-    if (!RoutablePlatform.ZEROX.supportsChain(chainId)) {
-      return resolve(undefined)
-    }
-
-    ZeroXTrade.bestTradeExactOut(currencyIn, currencyAmountOut, maximumSlippage)
-      .then(resolve)
-      .catch(error => {
-        errors.push(error)
-        resolve(undefined)
-        console.error(error)
-      })
-  })
-
-  // Gnosis Protocol V2
-  const gnosisProtocolTrade = new Promise<CoWTrade | undefined>(async resolve => {
-    if (!RoutablePlatform.GNOSIS_PROTOCOL.supportsChain(chainId as ChainId)) {
-      return resolve(undefined)
-    }
-
-    CoWTrade.bestTradeExactOut({
-      currencyAmountOut,
-      currencyIn,
-      maximumSlippage,
-      receiver,
-      user,
+        return undefined
+      }
     })
```

Comment:
This whole block is repeated above. I would advise creating helper function which would take care of both cases and which would ease the maintenance in the future considering we would have to update the logic on one place.